hubblestack/extmods/fileserver/s3fs.py

```diff
@@ -248,7 +246,7 @@ def serve_file(load, fnd):
     with hubblestack.utils.files.fopen(cached_file_path, 'rb') as fp_:
         fp_.seek(load['loc'])
         data = fp_.read(__opts__['file_buffer_size'])
-        if data and six.PY3 and not hubblestack.utils.files.is_binary(cached_file_path):
+        if data and not hubblestack.utils.files.is_binary(cached_file_path):
             data = data.decode(__salt_system_encoding__)
         if gzip and data:
             data = salt.utils.gzip_util.compress(data, gzip)
```

Comment:
Need to check for all such occurrence.
Should move under hubblestack.utils.gzip_util